utils/cli/cli.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+import functools
+import json
+import logging
+import requests
+import sys
+from io import BytesIO
+from PIL import Image
+from http.client import HTTPConnection
+from definition import parser, ResourceType
+log = logging.getLogger(__name__)
+
+
+class CLI():
+
+    def __init__(self, session, api):
+        self.api = api
+        self.session = session
+
+    def exception(func):
```

Comment:
Probably you want to define the method outside class `CLI`. It thinks that func argument is self.